### PR TITLE
Tracking/Phone Home/Upgrade notification

### PR DESCRIPTION
**Pull request by davecranwell**
Addresses https://github.com/torchbox/wagtail/issues/863

### CODE REVIEW

**Comment by davecranwell** [comment]
- [ ] Awaiting setup of https://releases.wagtail.io/latest.txt

Expected payload:
```
{
    'version' :'1.2.3'    // Version number. Can only contain numbers and decimal point.
    'url' : 'https://wagtail.io'  // Absolute URL to page/file containing release notes or actual package. It's up to you.
}
```

**Comment by tomdyson** [comment]
cloudfront setup done: https://releases.wagtail.io/latest.txt

**Comment by davecranwell** [comment]
@tomdyson: "XMLHttpRequest cannot load https://releases.wagtail.io/latest.txt. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:8111' is therefore not allowed access."

File: wagtail/wagtailadmin/views/home.py [hunk 1]
```diff
@@ -9,6 +9,20 @@
...
+
+    def __init__(self, request):
+        self.request = request
+
+    def render(self):
+        if self.request.user.is_superuser and getattr(settings, "WAGTAIL_ENABLE_TRACKING", True):
+            return render_to_string('wagtailadmin/home/upgrade_notification.html', {}, request=self.request)
+        else:
+            return ""
+
```
Comment (kaedroho) [inline wagtail/wagtailadmin/views/home.py]: Missing a space here

**Comment by davecranwell** [comment]
@tomdyson I guess you'd be best place to add steps to the release checklist https://github.com/torchbox/wagtail/wiki/Creating-a-new-Wagtail-release

**Comment by kaedroho** [comment]
Could you rebase/squash this?

**Comment by zerolab** [comment]
@davecranwell added notes for the release. Anything else left to do?

File: wagtail/wagtailadmin/templates/wagtailadmin/home.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {% extends "wagtailadmin/base.html" %}
-{% load gravatar static compress i18n %}
+{% load gravatar static compress i18n wagtailcore_tags%}
```
Comment (zerolab) [inline wagtail/wagtailadmin/templates/wagtailadmin/home.html]: add space after `wagtailcore_tags`

**Comment by zerolab** [comment]
One thing that popped in my mind when adding the latest.txt notes to https://github.com/torchbox/wagtail/wiki/Creating-a-new-Wagtail-release is that we have the LTS branch (0.8.x) and the latest stable. 

Should we concern ourselves with LTS at all? If so, what is the recommended approach? There are concerns about identifying which version to compare against.

The PyPi JSON has a `stable_version` key, yet I could not find anything suggesting how that can be set.

Feedback is welcome.

**Comment by davecranwell** [comment]
Having two `latest.txt` files (one "latest" one "stable") is one option. As is having both versions listed in a single file, however procedurally knowing which one to compare against is the main issue. Perhaps in addition to `__version__`, `wagtailcore/__init__.py` could also state whether it was a stable version or not?

Alternatively, `latest.txt` could itself dictated what the stable version was, which would allow us to change it on the fly without repackaging wagtail:

```
{
  "latest":"1.0.0",
  "stable_latest":"0.8.6",
  "stable_version_stub":"0.8"
}
```

**Comment by gasman** [comment]
Firstly, let's avoid using "stable" as a synonym for LTS - in all other contexts (such as git branch names) "stable" means any non-beta release. 1.0 final will be a "stable" release, but not (necessarily) a LTS one.

If we're going to cater for users following the LTS path, then I think we have to have something in the settings to indicate which upgrade path they want to follow - there's no way to do it just by comparing versions. Suppose the upgrade notification existed already in 0.8 - when 1.0 comes out, we can't automatically tell the difference between someone who should be offered an upgrade (because they want the latest and greatest) and someone who shouldn't (because they want to stick with 0.8 until the next LTS release).

**Comment by gasman** [comment]
How long does Cloudfront cache the file for? Would be good to mention that in the release checklist, so that we know when to start worrying if it hasn't updated yet... :-)

**Comment by tomdyson** [comment]
A day by default, but you can purge it through the horrible AWS interface.

**Comment by zerolab** [comment]
Had a chat with @gasman earlier today. The LTS issue can be tackled as part of work on 1.1 or 1.2.

**Comment by davecranwell** [comment]
https://releases.wagtail.io/latest.txt is showing access denied again. This appears to happen whenever someone touches the file so clearly the instructions are missing a step related to permissions,

**Comment by zerolab** [comment]
Updated the instructions.

**Comment by tomdyson** [comment]
Thanks, @zerolab. @ftarnell, any chance you could script this  - https://github.com/torchbox/wagtail/wiki/Creating-a-new-Wagtail-release#releaseswagtailio - up for us in Fabric / bash?

**Comment by ftarnell** [comment]
Wouldn't it be easier to host this ourselves...?

But yes, a shell script to upload a single file shouldn't be a problem.

**Comment by ftarnell** [comment]
Spec:

```
tom: something like `update_wagtail_release.sh --url="http://docs.wagtail.io/“ —version=1.1`
tom: should set public perms and send a purge request to the cloudfront cache
```

File: CHANGELOG.txt [hunk 3]
```diff
@@ -56,6 +56,7 @@ Changelog
 * Fix: Added validation to prevent pages being created with only whitespace characters in their title fields (Frank Wiles)
 
 
+
```
Comment (kaedroho) [inline CHANGELOG.txt]: Remove this line

File: wagtail/wagtailadmin/templates/wagtailadmin/home.html [hunk 4]
```diff
@@ -31,5 +31,4 @@
     {% else %}
         <p>{% trans "This is your dashboard on which helpful information about content you've created will be displayed." %}</p>
     {% endif %}
-
-{% endblock %}
+{% endblock %}
```
Comment (kaedroho) [inline wagtail/wagtailadmin/templates/wagtailadmin/home.html]: Missing newline

File: wagtail/wagtailadmin/templates/wagtailadmin/home.html [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {% extends "wagtailadmin/base.html" %}
-{% load gravatar static compress i18n %}
+{% load gravatar static compress i18n wagtailcore_tags %}
```
Comment (kaedroho) [inline wagtail/wagtailadmin/templates/wagtailadmin/home.html]: Not needed

File: wagtail/wagtailadmin/views/home.py [hunk 6]
```diff
@@ -9,6 +9,20 @@
...
+    def __init__(self, request):
+        self.request = request
+
+    def render(self):
+        if self.request.user.is_superuser and getattr(settings, "WAGTAIL_ENABLE_UPDATE_CHECK", True):
+            return render_to_string('wagtailadmin/home/upgrade_notification.html', {}, request=self.request)
+        else:
+            return ""
+
```
Comment (kaedroho) [inline wagtail/wagtailadmin/views/home.py]: Two spaces between classes

File: wagtail/wagtailadmin/static/wagtailadmin/js/tracking.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        try{
+            if(data.version && gtVersion(data.version, currentVersion)){
+                var $container = $('.panel-upgrade-notification')
+                $('.newversion', $container).html(data.version);
+                $('.releasenotes-link', $container).attr('href', data.url);
+                $container.show();
+            }
+        } catch (e){}
+    });
+});
```
Comment (kaedroho) [inline wagtail/wagtailadmin/static/wagtailadmin/js/tracking.js]: Missing newline

File: wagtail/wagtailadmin/static/wagtailadmin/js/tracking.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
+$(function(){
```
Comment (kaedroho) [inline wagtail/wagtailadmin/static/wagtailadmin/js/tracking.js]: Can we rename this file?

File: wagtail/wagtailadmin/static/wagtailadmin/js/tracking.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    function gtVersion(a, b) {
+        return cmpVersion(a, b) > 0;
+    }
+
+    var trackingUrl = "https://releases.wagtail.io/latest.txt";
+    var currentVersion = window.wagtailVersion;
+
+    $.getJSON(trackingUrl, function(data) {
+        try{
+            if(data.version && gtVersion(data.version, currentVersion)){
```
Comment (kaedroho) [inline wagtail/wagtailadmin/static/wagtailadmin/js/tracking.js]: Coding style here seems a bit inconsistent. This file may fail the javascript code check. Could you rebase this branch?

**Comment by kaedroho** [comment]
Please could you squash your commits then rebase it onto the latest master?

**Comment by kaedroho** [comment]
Just wrote this: https://gist.github.com/kaedroho/64b4ba554f650ec572f4
Feel free to stick it on a server somewhere run with a cron task (would be nicer than having to manually update this file on every single release)

**Comment by kaedroho** [comment]
Merged in c18bab66c61c7766deecb049af3d178edda39601

**Pull request closed without merging**